design/one-pager-moving-packages.md

```diff
@@ -0,0 +1,459 @@
+this design. It's covered by the [package `ImageConfig` proposal][imageconfig].
+
+## Proposal
+
+I propose we do two things:
+
+1. Allow packages to specify that they replace another package
+2. Allow packages to specify partially qualified dependencies
+
+These proposed changes correspond to the goals above. The first addresses
+moving dependencies, while the second addresses replicating dependencies.
+
+### Moving a Package
+
+I propose we support moving a package by adding a `replaces` stanza to package
+metadata.
+
+For example, assume the maintainers of `provider-family-aws` move it from
+`xpkg.upbound.io/upbound/provider-family-aws` to
+`xpkg.upbound.io/crossplane-contrib/provider-family-aws`. The maintainers would
+add the following `replaces` stanza to the moved provider:
+
+```yaml
+apiVersion: meta.pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-family-aws
+spec:
+  replaces:
+  - xpkg.upbound.io/upbound/provider-family-aws
+```
+
+This tells Crossplane that the `crossplane-contrib/provider-family-aws` package
+replaces the `upbound/provider-family-aws` package.
+
+When the revision controller encounters a package with a `replaces` stanza,
+it'll deactivate the replaced package(s) before installing the replacement
+package.
+
+After Crossplane deactivates the replaced package, it can install the
+replacement package without a conflict. A deactivated package [owns][owner-ref]
+but doesn't [control][controller-ref] its payload (e.g. a Provider's MR CRDs).
+This eliminates the controller reference conflict.
+
+A deactivated package exists, but in an inactive state. Specifically:
+
+* It appears under (e.g.) `kubectl get providers`
+* At least one revision appears under (e.g.) `kubectl get providerrevisions`
+* The package's `spec.revisionActivationPolicy` is set to `Manual`
+* All revisions have `spec.desiredState` set to `Inactive`
+* It has no runtime - e.g. no Deployment of a provider's runtime binary
+* It owns but doesn't control its package payload - e.g. a provider's MR CRDs
+
+Crucially, deactivating a package doesn't remove its payload. For example
+deactivating a provider doesn't remove its MR CRDs like uninstalling it would.
+The new package shares ownership (in the [owner reference][owner-ref] sense)
+with the replaced package, until the replaced package is manually uninstalled.
+This allows the package manager to seamlessly handover ownership and control of
+a package's payload.
+
+To deactivate a replaced packaged, the revision controller will:
+
+1. Map the replaced source to its active package revision using the Lock
+2. Map the package revision to its package using its owner reference
+3. Set the package's `spec.revisionActivationPolicy` to `Manual`
+4. Set the package revision's `spec.desiredState` to `Inactive`
+
+Once the replace source no longer appears in the Lock, the revision controller
+will proceed to install the new package.
```

Comment:
There's another issue now that we support upgrading a dependency.

When the package manager needs to update a replaced package to satisfy a new dependency constraint, it should update the replacement package - not the replaced one. The replaced package might not even really exist:

* Perhaps it was never installed in the first place
* Perhaps it was installed, then automatically deactivated, then uninstalled

Reply:
@negz can you help me understand why this is an issue? 
Let's say we installed `repoA/cool-pkg:v1.0.0` and replaced it with `repoB/cool-pkg:v1.0.0` and then we update a configuration which depends on `cool-pkg:v2.0.0`. Resolver controller needs to update the package which refers to `repoA/cool-pkg:v1.0.0` to `repoA/cool-pkg:v2.0.0` but the image will be pulled from `repoB/cool-pkg`(maybe we don't even have `v2.0.0` in repoA). I agree this is pretty confusing and not user-friendly. But is there a technical reason not implementing this flow? Do you think adding another field to package objects which shows the real(?) source will help the UX?